### PR TITLE
[FLINK-25657][Kinesis] Upgrade com.amazonaws:amazon-kinesis-client dependency from 1.14.1 to 1.14.7

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Update Amazon Kinesis Client dependency

## Brief change log

* Updated POM file
* Update NOTICE file

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-kinesis/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -6,7 +6,7 @@ The Apache Software Foundation (http://www.apache.org/).
 
 This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt) 
 
-- com.amazonaws:amazon-kinesis-client:1.14.1
+- com.amazonaws:amazon-kinesis-client:1.14.7
```
Comment (dannycranmer) [inline flink-connectors/flink-connector-kinesis/src/main/resources/META-INF/NOTICE:9]: Did you verify there are no new transient dependencies that need to be declared or excluded?

**Review by dannycranmer — APPROVED** [review]

File: flink-connectors/flink-connector-kinesis/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (MartijnVisser) [reply flink-connectors/flink-connector-kinesis/src/main/resources/META-INF/NOTICE:9]: Good point @dannycranmer 

When running `mvn compile dependency:tree` on `master`, this is returned for the Kinesis Connector:

```
[INFO] +- com.amazonaws:amazon-kinesis-client:jar:1.14.1:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.3.2:compile
[INFO] |  \- commons-logging:commons-logging:jar:1.1.3:compile
```

When validating it on this branch, it returns:

```
[INFO] +- com.amazonaws:amazon-kinesis-client:jar:1.14.7:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.3.2:compile
[INFO] |  +- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.13.0:compile
```

I'll add `com.fasterxml.jackson.core:jackson-core` to the NOTICE file.

**Comment by MartijnVisser** [comment]
@dannycranmer I've added the missing transitive dependencies, PTAL

**Review by dannycranmer — APPROVED** [review]

**Comment by MartijnVisser** [comment]
@dannycranmer Will you merge this?

**Comment by dannycranmer** [comment]
@MartijnVisser merged

**Pull request merged**